homeassistant/components/recorder/pool.py

```diff
@@ -20,6 +20,16 @@
 POOL_SIZE = 5
 
 
+def _raise_if_main_thread() -> None:
+    """Raise an exception if we are running in the main thread."""
+    if threading.current_thread() == threading.main_thread():
```

Comment:
Let's use the same approach as we use for our other guards: https://github.com/home-assistant/core/blob/dev/homeassistant/util/async_.py#L34-L36

([it's set here](https://github.com/home-assistant/core/blob/dev/homeassistant/core.py#L318))

Reply:
Actually, can we use the [`protect_loop`](https://github.com/home-assistant/core/blob/dev/homeassistant/util/async_.py#L168) or [`check_loop`](https://github.com/home-assistant/core/blob/dev/homeassistant/util/async_.py#L97) functions

Reply:
I couldn't use the decorator since it broke the inheritance but I was able to use `check_loop`